std/datetime/mod.ts

```diff
@@ -150,3 +149,129 @@ export function toIMF(date: Date): string {
+  return !(year % 400) || (year % 4 === 0 && year % 100 !== 0);
+}
+
+export type Unit =
+  | "miliseconds"
+  | "seconds"
+  | "minutes"
+  | "hours"
+  | "days"
+  | "weeks"
+  | "months"
+  | "quarters"
+  | "years";
+
+export type DifferenceFormat = {
+  miliseconds?: number;
+  seconds?: number;
+  minutes?: number;
+  hours?: number;
+  days?: number;
+  weeks?: number;
+  months?: number;
+  quarters?: number;
+  years?: number;
+};
+
+export type DifferenceOptions = {
+  units?: Unit[];
+};
+
+/**
+ * Calculate difference between two dates.
+ * @param from Year to calculate difference
+ * @param to Year to calculate difference with
+ * @param options Options for determining how to respond
+ *
+ * example :
+ *
+ * ```typescript
+ * datetime.difference(new Date("2020/1/1"),new Date("2020/2/2"),{ units : ["days","months"] })
+ * ```
+ */
+export function difference(
+  from: Date,
+  to: Date,
+  options?: DifferenceOptions
+): DifferenceFormat {
+  const uniqueUnits = options?.units
+    ? [...new Set(options?.units)]
+    : [
+        "miliseconds",
+        "seconds",
+        "minutes",
+        "hours",
+        "days",
+        "weeks",
+        "months",
+        "quarters",
+        "years",
+      ];
+
+  const bigger = from > to ? from : to;
+  const smaller = from > to ? to : from;
+  const differenceInMs = bigger.getTime() - smaller.getTime();
+
+  function calculateMonthDifference(biger: Date, smaller: Date): number {
+    const yearDiff = biger.getFullYear() - smaller.getFullYear();
+    const monthsDiff = biger.getMonth() - smaller.getMonth();
+    const calendarDiffrences = Math.abs(yearDiff * 12 + monthsDiff);
+    const compareResult = biger > smaller ? 1 : -1;
+    biger.setMonth(biger.getMonth() - compareResult * calendarDiffrences);
+    const isLastMonthNotFull =
+      biger > smaller ? 1 : -1 === -compareResult ? 1 : 0;
+    const months = compareResult * (calendarDiffrences - isLastMonthNotFull);
+    return months === 0 ? 0 : months;
+  }
+
+  const differences: DifferenceFormat = {};
+
+  for (let i = 0; i < uniqueUnits.length; i++) {
+    switch (uniqueUnits[i]) {
+      case "miliseconds":
+        differences.miliseconds = differenceInMs;
+        break;
+      case "seconds":
+        differences.seconds = Math.floor(differenceInMs / SECONDS);
+        break;
+      case "minutes":
+        differences.minutes = Math.floor(differenceInMs / MINUTES);
+        break;
+      case "hours":
+        differences.hours = Math.floor(differenceInMs / HOURS);
+        break;
+      case "days":
+        differences.days = Math.floor(differenceInMs / DAYS);
+        break;
+      case "weeks":
+        differences.weeks = Math.floor(differenceInMs / WEEKS);
+        break;
+      case "months":
+        differences.months = calculateMonthDifference(bigger, smaller);
+        break;
+      case "quarters":
+        const quarters =
+          (differences.months != undefined && differences.months / 4) ||
```

Comment:
it's safer to check whether a variable is defined like this `typeof differences.months !== 'undefined'` (https://stackoverflow.com/questions/3390396/how-to-check-for-undefined-in-javascript).

If you're intentionally using the casting brought by the loose inequality operator (as in checking whether it's null or undefined), I think it's a better idea to be more explicit about what you're checking